service/process.go

```diff
@@ -191,6 +191,12 @@ func (this *service) processIncoming(msg message.Message) error {
 func (this *service) processAcked(ackq *sessions.Ackqueue) {
 	for _, ackmsg := range ackq.Acked() {
 		// Let's get the messages from the saved message byte slices.
+
+		if len(ackmsg.Msgbuf) == 0 || len(ackmsg.Ackbuf) == 0 {
+			glog.Errorf("process/processAcked: Unable to decode new %s as ackmsg Msgbuf buf is %v or ackmsg Ackbuf buf is %v", ackmsg.Mtype, ackmsg.Msgbuf, ackmsg.Ackbuf)
```

Comment:
i/o timeout issues  is case by the server setTimeout , when you have a heartbeat or send message interval less than timeout,it should not apper 

the code here is just check the buffer len more than 0,that the message library 
miss 

you should not get the 0 buffer , so I fill it in ackqueue.go like below 

but if you get the 0 buffer , you should check the message that client send ,
or the server get [message](https://github.com/influxdata/surgemq/blob/master/service/process.go#L52)
, it may have a bug 

by the way , i did not use this in Industrial env